src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -2337,9 +2337,9 @@ private void OnTimeoutAsync(object state)
             }
         }
 
-        private bool OnTimeoutSync()
+        private bool OnTimeoutSync(bool asyncClose = false)
```

Comment:
NIT
```suggestion
        private bool OnTimeout(bool asyncClose = false)
```

Reply:
FYI - AsyncClose has nothing to do with OnTimeout being run synchronously. It's weird, but AsyncClose is specifically needed when OnTimeoutSync gets called from ReadAsyncCallback.